hkmc2/shared/src/main/scala/hkmc2/semantics/ucs/Desugarer.scala

```diff
@@ -505,22 +505,25 @@ class Desugarer(elaborator: Elaborator)(using Ctx, Raise, State, UnderCtx) exten
                     Split.Let(sym, callTupleGet(ref, -1 - lastIndex, sym), wrapInner(split))
                   (wrap, (sym, pat) :: matches)
             val lastMatches = reversedLastMatches.reverse
+            val sliceFn = if kw.isInstanceOf[Keyword.`..`.type] then tupleLazySlice else tupleSlice
```

Comment:
This is bug-prone. Next time these are refactored, it may silently start failign!
```suggestion
            val sliceFn = kw match
              case Keyword.`..` => tupleLazySlice
              case Keyword.`...` => tupleSlice
```